suite-common/wallet-core/src/fiat-rates/fiatRatesThunks.ts

```diff
@@ -31,37 +41,89 @@ export const updateTxsFiatRatesThunk = createThunk(
     ) => {
         if (txs?.length === 0 || isTestnet(account.symbol)) return;
 
-        const timestamps = txs.map(tx => tx.blockTime!);
-
         const isElectrumBackend = selectIsElectrumBackendSelected(getState(), account.symbol);
 
         try {
+            const timestamps = txs.map(tx => tx.blockTime);
+            const roundedTimestamps = roundTimestampsToNearestPastHour(timestamps as Timestamp[]);
+            const uniqueTimestamps = [...new Set(roundedTimestamps)];
+
             const results = await getFiatRatesForTimestamps(
                 { symbol: account.symbol },
-                timestamps,
+                uniqueTimestamps,
                 localCurrency,
                 isElectrumBackend,
             );
 
             if (results && 'tickers' in results) {
                 dispatch(
-                    transactionsActions.updateTransactionFiatRate(
-                        txs.map((tx, i) => ({
-                            txid: tx.txid,
-                            updateObject: { rates: results.tickers[i]?.rates },
-                            account,
-                            ts: Date.now(),
+                    fiatRatesActions.addFiatRatesForTimestamps({
+                        ticker: { symbol: account.symbol },
+                        localCurrency,
+                        rates: results.tickers.map((ticker, index) => ({
+                            rate: ticker?.rates[localCurrency],
+                            timestamp: uniqueTimestamps[index],
                         })),
-                    ),
+                    }),
                 );
             }
         } catch (error) {
             console.error(error);
         }
+
+        const groupedTokensTxs = groupTokensTransactionsByContractAddress(txs);
+
+        for (const token in groupedTokensTxs) {
```

Comment:
I wonder if we can introduce something like this:
![Screenshot from 2024-05-07 13-41-03](https://github.com/trezor/trezor-suite/assets/152899911/5d7cb01f-79b7-4702-aa8b-efa038db41d5)

To avoid casting the key every time we use it as here below:

`groupedTokensTxs[token as TokenAddress]`

@Nodonisko wdyt?

Reply:
The cast of the `token` is repeated 4 times below its definition here